rules/prebuilt_framework.bzl

```diff
@@ -0,0 +1,88 @@
+load("@build_bazel_rules_apple//apple:apple.bzl", "apple_dynamic_framework_import", "apple_static_framework_import")
+load("@build_bazel_rules_apple//apple/internal:apple_framework_import.bzl", "AppleFrameworkImportInfo")
+load("@build_bazel_rules_swift//swift/internal:providers.bzl", "SwiftUsageInfo")
+
+def apple_dynamic_framework_import_patched(name, **kwargs):
+    """Patches an apple_dynamic_framework_import target based on the problems reported in https://github.com/bazel-ios/rules_ios/issues/55
+
+    Args: same as the ones of apple_dynamic_framework_import
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.pop("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_dynamic_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = True,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def apple_static_framework_import_patched(name, **kwargs):
+    """Patches an apple_static_framework_import target based on the problems reported in https://github.com/bazel-ios/rules_ios/issues/55
+
+    Args: same as the ones of apple_static_framework_import
+    """
+
+    visibility = kwargs.pop("visibility", None)
+    tags = kwargs.get("tags", [])
+
+    legacy_target_label = "_" + name
+    apple_static_framework_import(
+        name = legacy_target_label,
+        tags = tags + ["manual"],
+        **kwargs
+    )
+
+    target_patcher(
+        name = name,
+        legacy_target = legacy_target_label,
+        isDynamic = False,
+        visibility = visibility,
+        tags = tags,
+    )
+
+def _target_patcher_impl(ctx):
+    old_objc_provider = ctx.attr.legacy_target[apple_common.Objc]
+
+    # Merge providers
+    objc_provider_fields = {
+        "providers": [old_objc_provider],
+        # Adding the module_map to the headers. No idea why this works, but this "hack" is based on:
```

Comment:
The "hack" is just a way to propagate the module map. `swift_rules` has a field in its provider called `swiftc_inputs` where you can pass extra dependencies to the compiler. Unfortunately the objc provider has no such field. The "hack" is to pass the module map as a header so that it makes it to the list of dependencies. Since it doesn't end in `.h` it gets ignored but it's still propagated as a dependency.

Reply:
I updated the comment with your information, thanks!